### ISSUE TITLE
engine/set-keymap-prefix not working in Emacs 29.0.50

### ISSUE THREAD

**Issue by grtcdr** [hrs/engine-mode#53]
Hi, 

I really enjoyed [your talk](https://www.youtube.com/watch?v=MBhJBMYfWUo) on this really clever mode. Thank you so much :heart:

Anyhow, after playing around with engine-mode's functionalities through `M-x`, I decided, like you mentioned (in the talk) to make use of the prefix key. But, unfortunately, `engine/set-keymap-prefix` doesn't seem to be working in my version of Emacs (29.0.5).

To anyone reading that's using a later version than usual, to set engine-mode's prefix you can do the following:

```elisp
(use-package engine-mode
  :ensure t
  :bind-keymap
  ("C-c /" . engine-mode-prefixed-map)
  :config
  (defengine github
             "https://github.com/search?q=%s"
             :keybinding "g"))
```

**Comment by hrs** [comment]
Oh hey, thanks @grtcdr!

I can't seem to reproduce this issue, though, using `29.0.50`.

Both of these methods seem to work on my machine. Do either of them work on yours?

1. Using `use-package`:

```emacs
(use-package engine-mode
  :ensure t

  :config
  (engine-mode t)
  (engine/set-keymap-prefix (kbd "C-c /"))
  (defengine github
    "https://github.com/search?q=%s"
    :keybinding "g"))
```

2. Loading the file manually:

```emacs
(load "path/to/engine-mode.el")
(require 'engine-mode)
(engine-mode t)

(engine/set-keymap-prefix (kbd "C-c /"))

(defengine github
  "https://github.com/search?q=%s"
  :keybinding "g")
```

If those aren't working, could you let me know if `engine/search-github` is still bound to the default binding of <kbd>C-x / g</kbd>?

(Incidentally, I actually like your use of `:bind-keymap` better! `engine/set-keymap-prefix` is an artifact of the dark ages before `use-package` became ubiquitous. :smile:)

**Comment by grtcdr** [comment]
Hello,

I tried the first snippet and it worked just fine, although I've downgraded to Emacs 28 for stability reasons a little while ago.

I haven't tried the second example, but I'm sure it'll work just as well since the expressions are no different.

I'm going to close this issue as it appears it's not reproducible.

Thank you for the help :smile:

> (Incidentally, I actually like your use of :bind-keymap better! engine/set-keymap-prefix is an artifact of the dark ages before use-package became ubiquitous. smile)

Oh, thanks! If you'd like, I could make a PR that includes a snippet for `use-package` and list it under https://github.com/hrs/engine-mode#changing-the-keymap-prefix.
